src/v0/helpers.ts

```diff
@@ -0,0 +1,229 @@
+    return 1;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(2))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(2))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(2)))
+  ) {
+    return 2;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(3))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(3))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(3)))
+  ) {
+    return 3;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(4))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(4))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(4)))
+  ) {
+    return 4;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(5))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(5))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(5)))
+  ) {
+    return 5;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(6))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(6))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(6)))
+  ) {
+    return 6;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(7))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(7))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(7)))
+  ) {
+    return 7;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(8))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(8))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(8)))
+  ) {
+    return 8;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(9))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(9))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(9)))
+  ) {
+    return 9;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(10))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(10))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(10)))
+  ) {
+    return 10;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(11))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(11))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(11)))
+  ) {
+    return 11;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(12))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(12))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(12)))
+  ) {
+    return 12;
+  } else if (
+    timestamp.equals(steward.timeLastCollected(BigInt.fromI32(42))) &&
+    tokenPrice.equals(steward.price(BigInt.fromI32(42))) &&
+    owner.equals(steward.currentPatron(BigInt.fromI32(42)))
+  ) {
+    return 42;
+  } else {
+    return -1; // a random non-released token -- this normally means the token was foreclosed or something like that
+  }
+}
+// A token would need to be set to the same price
+export function getTokenIdFromTimestamp(
```

Comment:
Yeah this doesn't work. The above function also broken foreclosures me thinks

Reply:
It works for the transactions that were made. But I don't understand what you mean exactly. Can you give more details?

Reply:
Sorry read my above comment - it follows on from that.